addons/controllers/calico/calicoconfig_controller.go

```diff
@@ -6,16 +6,34 @@ package controllers
 
 import (
 	"context"
+	"fmt"
+	"time"
 
+	"github.com/fatih/structs"
 	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	yaml "gopkg.in/yaml.v3"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/utils/pointer"
+	clusterapiv1beta1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	clusterapiutil "sigs.k8s.io/cluster-api/util"
+	clusterapipatchutil "sigs.k8s.io/cluster-api/util/patch"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 
+	"github.com/vmware-tanzu/tanzu-framework/addons/pkg/constants"
+	"github.com/vmware-tanzu/tanzu-framework/addons/pkg/util"
 	cniv1alpha1 "github.com/vmware-tanzu/tanzu-framework/apis/cni/v1alpha1"
 )
 
-// CalicoConfigReconciler reconciles a CalicoConfig object
+const contextTimeout = 30 * time.Second
```

Comment:
question: could this timeout shared by all config controllers or it is `CalicoConfig` only ?

Reply:
Do we need a timeout for controllers if all this is doing an k8s API request call. I think context passed by manager should be enough as is as it supports cancellation on `SIGHUP` or `SIGKILL`

Reply:
Ok, removed ti use the context passed by manager. Thanks for the comment.